sdk/instance/instance.go

```diff
@@ -0,0 +1,97 @@
+package instancesdk
+
+import (
+	"crypto/sha256"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/docker/docker/pkg/archive"
+	"github.com/mesg-foundation/core/container"
+	"github.com/mesg-foundation/core/database"
+	"github.com/mesg-foundation/core/instance"
+	"github.com/mesg-foundation/core/x/xos"
+	"github.com/mr-tron/base58"
+)
+
+// Instance exposes service instance APIs of MESG.
+type Instance struct {
+	container  container.Container
+	serviceDB  database.ServiceDB
+	instanceDB database.InstanceDB
+}
+
+// New creates a new Instance SDK with given options.
+func New(c container.Container, serviceDB database.ServiceDB, instanceDB database.InstanceDB) *Instance {
+	return &Instance{
```

Comment:
@krhubert https://github.com/mesg-foundation/engine/pull/1037#discussion_r291853480
> sdk should use the manager as a bridge to container package

Reply:
I don't really understand the meaning of your comment

Reply:
so instance shouldn't take container as a parameter (rather a manager).

This is why `manager` struct exists to use it. container is lets say "low level" package and should be used only by `manager`.

If you start to use container it will be all over the code. This was one of the reason why manager was created - to have api hiding implementation details.

Reply:
agree but the goal here is to slowly migrate the manager (that is in the service package) to this part and only the instance SDK will have this logic to manage anything related to the container (because only the instance needs to know about that).
So basically the manager will disappear and move in instanceSDK.

Also here this is not the instance model but the instance SDK/controller, the model will not know anything about the container. Only this instanceSDK should know about that (and the global SDK to pass the dependency only)